types/types.go

```diff
@@ -3,10 +3,14 @@
 
 package types
 
-import "errors"
+import (
+	"errors"
+
+	"github.com/hashicorp/raft"
+)
 
 var (
-	ErrNotFound = errors.New("log entry not found")
+	ErrNotFound = raft.ErrLogNotFound
```

Comment:
Could we add a comment about why it matters that this is exactly the same error? Something like "The raft library checks equality on the error to handle certain cases so we must return the same exact error as the raft library defined here.